norfair/metrics.py

```diff
@@ -0,0 +1,327 @@
+            index_position_on_this_document += 1
+        value_string = ""
+        while self.file[index_position_on_this_document].isdigit():
+            value_string = value_string + self.file[index_position_on_this_document]
+            index_position_on_this_document += 1
+        return int(value_string)
+
+def write_predictions(frame_number, objects=None, output_text_file=None):
+    '''
+    Write tracked object information on the output file (for this frame), in the format
+    frame_number, id, bb_left, bb_top, bb_width, bb_height, -1, -1, -1, -1
+    '''
+    # write tracked objects information in the output file
+    for t in range(len(objects)):
+        frame_str = str(int(frame_number))
+        id_str = str(int(objects[t].id))
+        bb_left_str = str((objects[t].estimate[0, 0]))
+        bb_top_str = str((objects[t].estimate[0, 1]))  # [0,1]
+        bb_width_str = str((objects[t].estimate[1, 0] - objects[t].estimate[0, 0]))
+        bb_height_str = str((objects[t].estimate[1, 1] - objects[t].estimate[0, 1]))
+        row_text_out = (
+            frame_str
+            + ","
+            + id_str
+            + ","
+            + bb_left_str
+            + ","
+            + bb_top_str
+            + ","
+            + bb_width_str
+            + ","
+            + bb_height_str
+            + ",-1,-1,-1,-1"
+        )
+        output_text_file.write(row_text_out)
+        output_text_file.write("\n")
+
+
+def process_text_file(text_file_path):
+    """
+    This function proccess detections and ground truth text files, ordering the data by frame, and making the box coordinates reference to corners positions.
+    """
+    matrix_data = np.loadtxt(text_file_path, dtype="f", delimiter=",")
+    row_order = np.argsort(matrix_data[:, 0])
+    matrix_data = matrix_data[row_order]
+    # coordinates refer to box corners
+    matrix_data[:, 4] = matrix_data[:, 2] + matrix_data[:, 4]
+    matrix_data[:, 5] = matrix_data[:, 3] + matrix_data[:, 5]
+    return matrix_data
+
+
+class PredictionsTextFile:
+    def __init__(self, input_path, save_path=".", information_file=None):
+
+        file_name = os.path.split(input_path)[1]
+
+        if information_file is None:
+            seqinfo_path = os.path.join(input_path, "seqinfo.ini")
+            information_file = InformationFile(file_path = seqinfo_path)
+
+        self.length = information_file.search(variable_name = "seqLength")
+
+        predictions_folder = os.path.join(save_path, "predictions")
+        if not os.path.exists(predictions_folder):
+            os.makedirs(predictions_folder)
+
+        out_file_name = os.path.join(predictions_folder, file_name + ".txt")
+        self.text_file = open(out_file_name, "w+")
+
+        self.frame_number = 1
+
+    def update_text_file(self, predictions):
+        write_predictions(
+            frame_number=self.frame_number, objects=predictions, output_text_file=self.text_file
+        )
```

Comment:
If we are only going to call this function this one time can't we just paste its contents here and delete the function entirely? The function `update_text_file ` does too little to justify having this piece of code abstracted away into a separate function.